sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1079,7 +1079,20 @@ case class GpuRegExpExtract(
       regexp: GpuScalar,
       idx: GpuScalar): ColumnVector = {
 
-    val groupIndex = idx.getValue.asInstanceOf[Int]
+    // When group index is 0, it means extract the entire pattern including those parts which
+    // don't belong to any group. For instance,
+    // regexp_extract('123abcEfg', '([0-9]+)[a-z]+([A-Z])', 0) => 123abcE
+    // regexp_extract('123abcEfg', '([0-9]+)[a-z]+([A-Z])', 1) => 123
+    // regexp_extract('123abcEfg', '([0-9]+)[a-z]+([A-Z])', 2) => E
+    //
+    // To support the full match (group index 0), we wrap a pair of parentheses on the original
+    // cudfRegexPattern.
+    val (extractPattern, groupIndex) = idx.getValue match {
+      case i: Int if i == 0 =>
+        ("(" + cudfRegexPattern + ")", 0)
```

Comment:
What if `i != 0`?

Reply:
> What if `i != 0`?

Then it will fall through to the following `case i` statement

Reply:
This is Scala's "Pattern Guard" feature - https://docs.scala-lang.org/tour/pattern-matching.html